js_03-04/js/script.js

```diff
@@ -1,39 +1,29 @@
-var domElements = {
-    div: 'div',
-    h3: 'h3',
-    ol: 'ol',
-    ul: 'ul',
-    li: 'li',
-    input: 'input',
-    label: 'label',
-    button: 'button'       
-};
-
 var pageObj = {
 
-    insertNewDomElement: function(whereToInsert, tagName, cssClass) {
-        // You can send more than 3 arguments during call insertNewDomElement() function
-        // 4-th argument is Object and means that the dom element has additional params 
-        
-        var parentArray = document.querySelectorAll(whereToInsert);
+    insertNewDomElement: function( insertedObj ) {
 
+        var parentArray = document.querySelectorAll(insertedObj.parent);
 
         for ( var i = 0; i < parentArray.length; i++ ) {
-            var child = document.createElement(tagName);
+            var child = document.createElement(insertedObj.childTag);
 
-            if (cssClass !== '') {
-                child.classList.add(...cssClass);
+            if (insertedObj.childClass) {
+                child.classList.add(...insertedObj.childClass);
             }
 
-            if (arguments.length > 3) {
-                for ( var key in arguments[3] ) {
-                    child[key] = arguments[3][key];
+                if ( Object.keys(insertedObj).length > 3 ) {
+                    var j = 3;
+
+                    while ( j < Object.keys(insertedObj).length ) {
+                        var key = Object.keys(insertedObj)[j];
+                        child[key] = insertedObj[key];
+                        j++;
+                    }
```

Comment:
please check this method
https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Object/entries#%D0%9F%D1%80%D0%B8%D0%BC%D0%B5%D1%80%D1%8B
maybe it will be more handy for you

Reply:
@salieri328 please take a look at the new version https://github.com/oonufrienko/oonufrienko.github.io/pull/7 of implementation. I didn't use either `Object.keys` or `Object.entries` functions. Because the usage of `Object.entries` function means that you should enumerate all the values within result array. So, I just added two `if` statements and the code became more simple and readable.